test/10-verify.spec.js

```diff
@@ -809,6 +809,40 @@ for(const [version, mockCredential] of versionedCredentials) {
           .contain('"issuer" must be a URI');
       });
 
+      it('should reject credentialStatus that is not a URI', () => {
+        const credential = mockCredential();
+        credential.credentialStatus = {
+          id: 'not-a-url',
+          type: 'urn:type'
+        };
+        let error;
+        try {
+          vc._checkCredential({credential});
+        } catch(e) {
+          error = e;
+        }
+        should.exist(error,
+          'Should throw error when "evidence" is not a URI');
+        error.should.be.instanceof(TypeError);
+        error.message.should
+          .contain('"credentialStatus.id" must be a URI');
+      });
+
+      it('should accept "credentialStatus" with no "id"', () => {
```

Comment:
is this 1.0 or 2.0? in 2.0 I'm showing: https://w3c.github.io/vc-data-model/#status


> credentialStatus
> If present, the value of the credentialStatus [property](https://w3c.github.io/vc-data-model/#dfn-property) MUST include the following:
> id [property](https://w3c.github.io/vc-data-model/#dfn-property), which MUST be a [URL](https://w3c.github.io/vc-data-model/#dfn-url) which MAY be dereferenced.
> type [property](https://w3c.github.io/vc-data-model/#dfn-property), which expresses the [credential](https://w3c.github.io/vc-data-model/#dfn-credential) status type.

so it should not accept a credentialStatus with no id. @davidlehn

Reply:
also checked and in 1.0 `credentialStatus.id` must be a url and present:

https://www.w3.org/TR/vc-data-model/#status

> credentialStatus
> If present, the value of the credentialStatus [property](https://www.w3.org/TR/vc-data-model/#dfn-property) MUST include the following:
> id [property](https://www.w3.org/TR/vc-data-model/#dfn-property), which MUST be a [URI](https://www.w3.org/TR/vc-data-model/#dfn-uri).
> type [property](https://www.w3.org/TR/vc-data-model/#dfn-property), which expresses the [credential](https://www.w3.org/TR/vc-data-model/#dfn-credential) status type (also referred to as the [credential](https://www.w3.org/TR/vc-data-model/#dfn-credential) status method). It is expected that the value will provide enough information to determine the current status of the [credential](https://www.w3.org/TR/vc-data-model/#dfn-credential) and that machine readable information needs to be retrievable from the URI. For example, the object could contain a link to an external document noting whether or not the [credential](https://www.w3.org/TR/vc-data-model/#dfn-credential) is suspended or revoked.

Reply:
to summarize: in vc 2.0 credential status id is becoming optional because specs that define credential status(es) don't necessarily want to make information like id available. So in the higher level spec (VC 2.0) id will become optional and then depending on the credential status spec id will be required or not.